src/core/tsi/ssl_transport_security.c

```diff
@@ -1282,8 +1282,10 @@ tsi_result tsi_create_ssl_client_handshaker_factory(
   *factory = NULL;
   if (pem_root_certs == NULL) return TSI_INVALID_ARGUMENT;
 
-  ssl_context = SSL_CTX_new(TLSv1_2_method());
-  if (ssl_context == NULL) {
+  ssl_context = SSL_CTX_new(TLS_method());
+  if (ssl_context == NULL ||
+      !SSL_CTX_set_min_proto_version(ssl_context, TLS1_2_VERSION) ||
+      !SSL_CTX_set_max_proto_version(ssl_context, TLS1_2_VERSION)){
```

Comment:
space between `)` and `{` 
Should be taken care of by the clang-format command:
`./tools/distrib/clang_format_code.sh --use-docker`